File: PureWebUI/PureWebUI/AppStateViewController.m
//  AppStateViewController.m
//
//  Created by Chris Burns on 2/10/14.
//  Copyright 2012 Calgary Scientific Inc. All rights reserved.


#import "AppStateViewController.h"

#import <PureWeb/PureWeb.h>

@interface AppStateViewController ()

@property (strong, nonatomic) IBOutlet UIWebView *appStateWebView;

@end

@implementation AppStateViewController

- (void)viewDidLoad
{
    NSString *xml = [NSString stringWithFormat:@"<?xml version=\"1.0\" encoding=\"UTF-8\" ?>%@",
                     [[[PWFramework sharedInstance].state.stateManager getTree:@"/"] description]];
    
    
    [self.appStateWebView loadData:[xml dataUsingEncoding:NSUTF8StringEncoding]
                          MIMEType:@"text/plain"
                  textEncodingName:@"utf-8"
                           baseURL:[NSURL URLWithString:@"http://"]];
    
    
    [super viewDidLoad];
}



@end

File: Swift-Asteroids/Asteroids/Asteroids/AppStateViewController.swift
//
//  AppStateViewController.swift
//  Asteroids
//
//  Created by Jonathan Neitz on 2016-02-16.
//  Copyright © 2016 Calgary Scientific Inc. All rights reserved.
//

import UIKit
import PureWeb

class AppStateViewController: UIViewController {

    @IBOutlet weak var appStateWebView: UIWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        let description = PWFramework.sharedInstance().state().stateManager.getTree("/").description;
        
        let xml = "<?xml version=\"1.0\"encoding=\"UTF-8\" ?>\(description)"
        
        if let data = xml.data(using: String.Encoding.utf8) {
            
            appStateWebView.load(data, mimeType:"text/plain", textEncodingName:"utf-8", baseURL: URL( string:"http://")! );
        }
    }

}
